lib/bigtable/index.js

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module bigtable
+ */
+
+'use strict';
+
+var googleProtoFiles = require('google-proto-files');
+var nodeutil = require('util');
+var format = require('string-format-obj');
+
+/**
+ * @private
+ */
+var Table = require('./table.js');
+
+/**
+ * @private
+ */
+var GrpcService = require('../common/grpc-service.js');
+
+/**
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ *
+ */
+function BigTable(options) {
+  if (!(this instanceof BigTable)) {
+    options = util.normalizeArguments(this, options);
+    return new BigTable(options);
+  }
+
+  this.clusterId = options.clusterId;
+  this.projectId = options.projectId;
+  this.zoneId = options.zoneId;
+
+  this.clusterName = format(
+    'projects/{project}/zones/{zone}/clusters/{cluster}', {
+      project: this.projectId,
+      zone: this.zoneId,
+      cluster: this.clusterId
+    });
+
+  var config = {
+    baseUrl: 'bigtable.googleapis.com',
+    service: 'bigtable',
+    apiVersion: 'v1',
+    protoServices: {
+      BigtableService: googleProtoFiles.bigtable.v1,
+      BigtableTableService: {
```

Comment:
Should we just change all of our config.protoServices[service] to be an object with a path? I'm okay either way, just wondering if you think being more explicit with it would make it more obvious.